### PR TITLE
[node/fs] fix return and options parameter types for fs.watchFile; add class fs.StatWatcher

### PR DESCRIPTION
**Pull request by emarteca**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:
If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [URL here](https://nodejs.org/api/fs.html#fswatchfilefilename-options-listener)

Note: in fixing the return type for `fs.watchFile` I had to add a representation of the class `fs.StatWatcher`: [docs here](https://nodejs.org/api/fs.html#class-fsstatwatcher)

### CODE REVIEW

File: types/node/fs.d.ts [hunk 1]
```diff
@@ -2788,17 +2811,18 @@ declare module 'fs' {
         filename: PathLike,
         options:
             | {
+                  bigint?: boolean | undefined;
```
Comment (weswigham) [inline types/node/fs.d.ts:2814]: The `bigint` option being set changes the types in the `Stats` objects to be `bigint`s - shouldn't the types be updated to respect that?

Comment (emarteca) [reply types/node/fs.d.ts]: Yes, good catch! Just committed a fix.

**Comment by peterblazejewicz** [comment]
@emarteca please update from master and push changes, there were changes in other DT package that is part of Node ecosystem, impacting tests here. Thx!

**Comment by emarteca** [comment]
I just did merge master, the failing tests seem to be caused by `ari-client` and `hapi-pino`. I had the same issue yesterday with the DT master so I'm not sure what's going on.

**Comment by peterblazejewicz** [comment]
you're correct. Fixing CI resolutions: #57241

**Comment by emarteca** [comment]
Sweet, thanks! Looks like it fixed my issue :raised_hands:

File: types/node/fs.d.ts [hunk 2]
```diff
@@ -260,6 +260,29 @@ declare module 'fs' {
          */
         readSync(): Dirent | null;
     }
+    /**
+     * Class: fs.StatWatcher
+     * @since v14.3.0, v12.20.0
+     * Extends `EventEmitter`
+     * A successful call to {@link watchFile} method will return a new fs.StatWatcher object.
+     */
+    export class StatWatcher extends EventEmitter {
```
Comment (SimonSchick) [inline types/node/fs.d.ts:269]: It looks like this class does not exist, please make it an interface instead.

File: types/node/fs.d.ts [hunk 3]
```diff
@@ -2784,21 +2807,34 @@ declare module 'fs' {
...
+              })
             | undefined,
         listener: (curr: Stats, prev: Stats) => void
-    ): void;
+    ): StatWatcher;
+    export function watchFile(
+        filename: PathLike,
+        options:
+            | (WatchFileOptions & {
+                  bigint?: true;
```
Comment (SimonSchick) [inline types/node/fs.d.ts:2828]: iirc `bigint` must be explicitly true here.

**Review by SimonSchick — CHANGES_REQUESTED** [review]

File: types/node/fs.d.ts (hunk 2 above)
Comment (SimonSchick) [reply types/node/fs.d.ts:269]: To clarify:

Comment (emarteca) [reply types/node/fs.d.ts]: You're right, good catch!

File: types/node/fs.d.ts (hunk 3 above)
Comment (emarteca) [reply types/node/fs.d.ts]: Yes good catch, that was a typo on my part

File: types/node/fs.d.ts [hunk 4]
```diff
@@ -260,6 +260,29 @@ declare module 'fs' {
...
         readSync(): Dirent | null;
     }
+    /**
+     * Class: fs.StatWatcher
+     * @since v14.3.0, v12.20.0
+     * Extends `EventEmitter`
+     * A successful call to {@link watchFile} method will return a new fs.StatWatcher object.
+     */
+    export interface StatWatcher extends EventEmitter {
+        /**
```
Comment (SimonSchick) [inline types/node/fs.d.ts:270]: Also sorry I missed this, what are the available events on this emitter?

Comment (emarteca) [reply types/node/fs.d.ts:270]: Good question -- from [the documentation](https://nodejs.org/api/fs.html#class-fsstatwatcher) it doesn't look like there are any on the base class.

Comment (emarteca) [reply types/node/fs.d.ts:270]: [Looking at the implementation of `StatWatcher` in node](https://github.com/nodejs/node/blob/340b770d3f4c3b50a5d36c0776afd63d9b4f22de/lib/internal/fs/watchers.js#L68), it also doesn't look like there are any events on this emitter.

**Review by gabritto — APPROVED** [review]
thanks! 😊

**Pull request merged**